### ISSUE TITLE
How to implement multiple optimizers based on epoch number?

### ISSUE THREAD

**Issue by bartmch** [PyTorchLightning/pytorch-lightning#4834]
I want to use optimizer_1 until epoch 100, and then use optimizer_2. The issue is that my `training_epoch_end` step yields 0 or nan outputs between 1st and 2d epoch. My guess is that this could be a wrong multi-optimizer set up.
In my `configure_optimizers` I do:
![image](https://user-images.githubusercontent.com/10415569/100080657-87cc0e00-2e5f-11eb-91f6-296e631a49db.png)
In my `optimizer_step` I do:
![image](https://user-images.githubusercontent.com/10415569/100080753-a4684600-2e5f-11eb-9513-61c547ee47f0.png)
In my `training_step` I just give `optimizer_idx` but don't know what to do with it (this is maybe the issue?):
![image](https://user-images.githubusercontent.com/10415569/100080889-cc57a980-2e5f-11eb-95f4-07acf306c3e5.png)
With `_step` being:
![image](https://user-images.githubusercontent.com/10415569/100080923-db3e5c00-2e5f-11eb-8a51-172bc179e6fa.png)

I'm getting the error in `training_epoch_end`:
![image](https://user-images.githubusercontent.com/10415569/100081033-fc06b180-2e5f-11eb-825e-6d9777e7f58d.png)
With `_epoch_end` being:
![image](https://user-images.githubusercontent.com/10415569/100081079-0b85fa80-2e60-11eb-92fc-4547a1c595b2.png)
The values of the list "outputs" are containing zeros and nans - If I just use one optimizer, then everything works perfectly fine. Any off-topic feedback on my code is more than welcome :)

**Comment by rohitgr7** [comment]
check the example here: https://github.com/PyTorchLightning/pytorch-lightning/issues/4815
might help

**Comment by tchaton** [comment]
From Lightning Documentation : https://pytorch-lightning.readthedocs.io/en/stable/optimizers.html?highlight=scheduler#learning-rate-scheduling

```
# no LR scheduler
def configure_optimizers(self):
   return Adam(...)

# Adam + LR scheduler
def configure_optimizers(self):
   optimizer = Adam(...)
   scheduler = LambdaLR(optimizer, ...)
   return [optimizer], [scheduler]

# The ReduceLROnPlateau scheduler requires a monitor
def configure_optimizers(self):
   return {
       'optimizer': Adam(...),
       'lr_scheduler': ReduceLROnPlateau(optimizer, ...),
       'monitor': 'metric_to_track'
   }

# Two optimizers each with a scheduler
def configure_optimizers(self):
   optimizer1 = Adam(...)
   optimizer2 = SGD(...)
   scheduler1 = LambdaLR(optimizer1, ...)
   scheduler2 = LambdaLR(optimizer2, ...)
   return [optimizer1, optimizer2], [scheduler1, scheduler2]

# Alternatively
def configure_optimizers(self):
   optimizer1 = Adam(...)
   optimizer2 = SGD(...)
   scheduler1 = ReduceLROnPlateau(optimizer1, ...)
   scheduler2 = LambdaLR(optimizer2, ...)
   return (
       {'optimizer': optimizer1, 'lr_scheduler': scheduler1, 'monitor': 'metric_to_track'},
       {'optimizer': optimizer2, 'lr_scheduler': scheduler2},
   )

# Same as above with additional params passed to the first scheduler
def configure_optimizers(self):
   optimizers = [Adam(...), SGD(...)]
   schedulers = [
      {
         'scheduler': ReduceLROnPlateau(optimizers[0], ...),
         'monitor': 'metric_to_track',
         'interval': 'epoch',
         'frequency': 1,
         'strict': True,
      },
      LambdaLR(optimizers[1], ...)
   ]
   return optimizers, schedulers
```
